components/input/gpio/gpio.go

```diff
@@ -254,17 +258,16 @@ func (c *Controller) sendConnectionStatus(ctx context.Context, connected bool) {
 	}
 }
 
-func (c *Controller) newButton(ctx context.Context, brd board.Board, intName string, cfg ButtonConfig) error {
+func (c *Controller) newButton(ctx context.Context, brd board.Board, interrupt board.DigitalInterrupt, cfg ButtonConfig) error {
 	tickChan := make(chan board.Tick)
-	err := brd.StreamTicks(ctx, []string{intName}, tickChan, nil)
+	err := brd.StreamTicks(ctx, []board.DigitalInterrupt{interrupt}, tickChan, nil)
 	if err != nil {
 		return errors.Wrap(err, "error getting digital interrupt ticks")
 	}
 
 	c.activeBackgroundWorkers.Add(1)
 	utils.ManagedGo(func() {
-		// Remove the callbacks added by the interrupt stream once we are done.
-		defer utils.UncheckedErrorFunc(func() error { return board.RemoveCallbacks(brd, []string{intName}, tickChan) })
+		defer interrupt.RemoveCallback(tickChan)
```

Comment:
remember we're goign to remove these functions from the digital interrupt interface as per the scope doc, so we shouldn't increase use of them from the main digital interrupt interface.

Reply:
Rand is right, but removing their use is a separate task for a separate PR. If it's a lot of work right now to fix this, just leave it as-is and fix it later.